### PR TITLE
feat: implement user task search transformers

### PR DESCRIPTION
**Pull request by nathansandi**
## Description

Create the User Task Filter/Query on Camunda Service 

Unit tests available on this Pull Request: https://github.com/camunda/camunda/pull/19434

Note:
This filter will only contemplate User Task, to filter user task variables, this logic will be handled on the endpoint business logic for the performance reasons, only we will have the variable service, and we can divide the search by the tasks that are running vs. tasks that are completed. 

The variable service is available on this  Pull Request: https://github.com/camunda/camunda/pull/19385

## How to test? 

Example of Java Application
```
public class TestApplication {

  public static void main(final String[] args) {
    final var serverUrl = "http://localhost:9200";
    final var restClient = RestClient.builder(HttpHost.create(serverUrl)).build();
    final var transport = new RestClientTransport(restClient, new JacksonJsonpMapper());
    final var esClient = new ElasticsearchClient(transport);

    final var dataStoreClient = new ElasticsearchSearchClient(esClient);
    final var camundaServices = new CamundaServices(dataStoreClient);
    

    final SearchQueryResult<UserTaskEntity> result =
        camundaServices
            .userTaskServices()
            .search(
                (q) ->
                    q.page((p) -> p.size(100))
                        .filter((p) -> p.taskStates("COMPLETED"))
                        .sort((s) -> s.startDate().desc()));

    result.items().stream().forEach(System.out::println);

    System.exit(-1);
  }
}
```

## Related issues

closes #https://github.com/camunda/camunda/issues/19210

### CODE REVIEW

File: service/src/main/java/io/camunda/service/search/sort/UserTaskSort.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  public static UserTaskSort of(final Function<Builder, ObjectBuilder<UserTaskSort>> fn) {
+    return SortOptionBuilders.userTask(fn);
+  }
+
+  public static final class Builder extends AbstractBuilder<Builder>
+      implements ObjectBuilder<UserTaskSort> {
+
+    public Builder startDate() {
```
Comment (romansmirnov) [inline service/src/main/java/io/camunda/service/search/sort/UserTaskSort.java:29]: I haven't checked the specification or the v1 endpoints; in the filter, I can filter by `creationDate()` but sorting happens by `startDate()`, can this be aligned:

```suggestion
    public Builder creationDate() {
```

File: service/src/main/java/io/camunda/service/search/sort/UserTaskSort.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      currentOrdering = new FieldSorting("creationTime", null);
+      return this;
+    }
+
+    public Builder endDate() {
```
Comment (romansmirnov) [inline service/src/main/java/io/camunda/service/search/sort/UserTaskSort.java:34]: The same as above:

```suggestion
    public Builder completionDate() {
```

File: service/src/main/java/io/camunda/service/transformers/filter/UserTaskFilterTransformer.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    return null;
+  }
+
+  private SearchQuery getCanceledQuery(final boolean canceled) {
+    if (canceled) {
+      return term("state", "CANCELED");
+    }
+
+    return null;
+  }
```
Comment (romansmirnov) [inline service/src/main/java/io/camunda/service/transformers/filter/UserTaskFilterTransformer.java:174]: These methods are not used, and the todo is unclear to me. Can it be deleted for the time being?

File: service/src/main/java/io/camunda/service/search/filter/UserTaskFilter.java [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+    }
+
+    public Builder taskDefinitionIds(final String value, final String... values) {
+      return taskDefinitionIds(collectValues(value, values));
+    }
+
+    public Builder taskDefinitionIds(final List<String> values) {
+      taskDefinitionIds = addValuesToList(taskDefinitionIds, values);
+      return this;
+    }
```
Comment (romansmirnov) [inline service/src/main/java/io/camunda/service/search/filter/UserTaskFilter.java:76]: Should it be `userTaskDefinitionIds()`?

File: service/src/main/java/io/camunda/service/search/filter/UserTaskFilter.java [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+
+    public Builder taskStates(final String value, final String... values) {
+      return taskStates(collectValues(value, values));
+    }
+
+    public Builder taskStates(final List<String> values) {
+      taskStates = addValuesToList(taskStates, values);
+      return this;
+    }
```
Comment (romansmirnov) [inline service/src/main/java/io/camunda/service/search/filter/UserTaskFilter.java:103]: Should it be `userTaskState()`?

**Review by romansmirnov — CHANGES_REQUESTED** [review]
@nathansandi, nice to see the first implementation using that new infrastructure.

* I added some smaller comments
* I added a commit to clean up the `UserTaskFilter` (nothing big 😄 )

File: service/src/main/java/io/camunda/service/transformers/filter/UserTaskFilterTransformer.java (hunk 3 above)
Comment (nathansandi) [reply service/src/main/java/io/camunda/service/transformers/filter/UserTaskFilterTransformer.java:174]: True, I will delete for now :)

**Review by romansmirnov — APPROVED** [review]
@nathansandi, thanks for applying the review feedback.

Just a quick feedback, to ensure a smooth review process, please apply the feedback in new commits, and not amending them. That way, it easier for the reviewer to review the additional changes:

> If there are changes requested, address them in a new commit.

(cf. https://github.com/camunda/camunda/blob/main/CONTRIBUTING.md#creating-a-pull-request)

**Pull request merged**